sandbox/modules/identity/src/main/java/org/opensearch/identity/configuration/ConfigurationRepository.java

```diff
@@ -313,4 +337,43 @@ private Map<CType, SecurityDynamicConfiguration<?>> validate(Map<CType, Security
 
         return conf;
     }
+
+    public void initializeExtensionsSecurity() {
+        SecurityDynamicConfiguration<ExtensionSecurityConfig> extensionsSecurityConfig = SecurityDynamicConfiguration.empty();
+        Map<String, DiscoveryExtensionNode> extensionIdMap = IdentityPlugin.GuiceHolder.getExtensionsManager().getExtensionIdMap();
+        if (extensionIdMap != null) {
+            for (String extensionId : extensionIdMap.keySet()) {
+                ExtensionSecurityConfig es = new ExtensionSecurityConfig();
+                es.setSigningKey(ExtensionsSecretsGenerator.generateSigningKey());
+                extensionsSecurityConfig.putCObject(extensionId, es);
+            }
+
+            String configType = CType.EXTENSIONSECURITY.toLCString();
+
+            try {
+                final IndexRequest indexRequest = new IndexRequest(this.identityIndex).id(configType)
+                    .opType(DocWriteRequest.OpType.INDEX)
+                    .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
+                    .source(configType, XContentHelper.toXContent(extensionsSecurityConfig, XContentType.JSON, false));
+                IndexResponse response = client.index(indexRequest).actionGet();
+                final String res = response.getId();
+
+                if (!configType.equals(res)) {
```

Comment:
Are you just putting this here to be extra safe, or did you run into an issue where this was happening? Is it a response timing thing or a bad parse?

Reply:
I'm not sure what all of the overriden methods are for. I see the same methods in `OpenSearchSecurityPlugin` in the security plugin as well.